src/app/models/payment.server.model.ts

```diff
@@ -90,6 +91,22 @@ const PaymentSchema = new Schema<IPaymentSchema, IPaymentModel>(
         },
       },
     },
+
+    products: {
+      type: [
+        {
+          selected: {
+            type: Boolean,
+            required: true,
+          },
+          quantity: {
+            type: Number,
+            required: true,
+          },
+          data: ProductItemSchema,
```

Comment:
In spirit by @wanlingt 

ProductItemSchema is the schema for `ProductItem`. But actually this is defining the schema for `Product`. Might want to consider renaming this to `ProductSchema`

Reply:
1d20ad37